data/src/main/java/com/microsoft/azure/kusto/data/auth/UserPromptTokenProvider.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.kusto.data.auth;
+
+import com.microsoft.aad.msal4j.IAccount;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.InteractiveRequestParameters;
+import com.microsoft.aad.msal4j.PublicClientApplication;
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class UserPromptTokenProvider extends PublicAppTokenProviderBase {
+    private static final int USER_PROMPT_TIMEOUT_MS = 120 * 1000;
+    private static final URI redirectUri;
+
+    static {
+        URI tmp;
+        try {
+            tmp = new URI("http://localhost");
+        } catch (URISyntaxException e) { // This cannot happen, but is necessary to allow for a static final variable whose instantiation can throw an exception
+            tmp = null;
+        }
+        redirectUri = tmp;
+    }
+
+    private final String usernameHint;
+
+    UserPromptTokenProvider(@NotNull String clusterUrl, String authorityId) throws URISyntaxException {
+        this(null, clusterUrl, authorityId);
+    }
+
+    UserPromptTokenProvider(String usernameHint, @NotNull String clusterUrl, String authorityId) throws URISyntaxException {
+        super(clusterUrl, authorityId);
+        this.usernameHint = usernameHint;
+
+        try {
+            clientApplication = PublicClientApplication.builder(CLIENT_ID).authority(aadAuthorityUrl).build();
+        } catch (MalformedURLException e) {
+            throw new URISyntaxException(aadAuthorityUrl, ERROR_INVALID_AUTHORITY_URL);
+        }
+    }
+
+    @Override
+    protected IAuthenticationResult acquireNewAccessToken() throws DataServiceException, DataClientException {
+        IAuthenticationResult result;
+        try {
+            // This is the only auth method that allows the same application to be used for multiple distinct accounts, so reset account cache between sign-ins
+            clientApplication = PublicClientApplication.builder(CLIENT_ID).authority(aadAuthorityUrl).build();
+            CompletableFuture<IAuthenticationResult> future = clientApplication.acquireToken(InteractiveRequestParameters.builder(redirectUri).scopes(scopes).loginHint(usernameHint).build());

```

Comment:
are you sure this is ok if the hint is not initialized?

Reply:
Yes - I tested it.

Reply:
Default = null or comment then

Reply:
Same as comment above